File: java/src/main/java/com/sopra/Factory/model/Salle.java
package com.sopra.Factory.model;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("salle")
public class Salle extends RessourceMaterielle{
	
	@Column(name = "nb_personne")
	private Integer nbPersonne;
	
	@OneToMany(mappedBy="salle")
	//@JsonView(JsonViews.SalleByIdWithCursus.class)
    private Set<Cursus> cursus;
	
	

	
	
	
	public Set<Cursus> getCursus() {
		return cursus;
	}


	public void setCursus(Set<Cursus> cursus) {
		this.cursus = cursus;
	}


	public Salle() {
		super();
	}

	
	public Integer getNbPersonne() {
		return nbPersonne;
	}

	public void setNbPersonne(Integer nbPersonne) {
		this.nbPersonne = nbPersonne;
	}
	
	
	

}

File: angular/src/app/model/RessourceMaterielleHeritage/salle.ts
import {RessourceMaterielle} from '../ressourceMaterielle';
import {Cursus} from '../cursus';

export class Salle extends RessourceMaterielle {
  constructor(private _nbPersonne: number, private _cursus: Cursus[]) {
    super();
  }

  get nbPersonne(): number {
    return this._nbPersonne;
  }

  set nbPersonne(value: number) {
    this._nbPersonne = value;
  }

  get cursus(): Cursus[] {
    return this._cursus;
  }

  set cursus(value: Cursus[]) {
    this._cursus = value;
  }
}
